space/space_test.go

```diff
@@ -6,248 +6,260 @@ import (
 
 	"context"
 
-	"github.com/fabric8-services/fabric8-wit/account"
 	"github.com/fabric8-services/fabric8-wit/errors"
-	"github.com/fabric8-services/fabric8-wit/gormsupport/cleaner"
 	"github.com/fabric8-services/fabric8-wit/gormtestsupport"
 	"github.com/fabric8-services/fabric8-wit/resource"
 	"github.com/fabric8-services/fabric8-wit/space"
-
+	testsupport "github.com/fabric8-services/fabric8-wit/test"
 	tf "github.com/fabric8-services/fabric8-wit/test/testfixture"
-	errs "github.com/pkg/errors"
 	uuid "github.com/satori/go.uuid"
-	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/stretchr/testify/suite"
 )
 
-var testSpace string = uuid.NewV4().String()
-var testSpace2 string = uuid.NewV4().String()
-
 func TestRunRepoBBTest(t *testing.T) {
 	suite.Run(t, &repoBBTest{DBTestSuite: gormtestsupport.NewDBTestSuite("../config.yaml")})
 }
 
 type repoBBTest struct {
 	gormtestsupport.DBTestSuite
-	repo         space.Repository
-	testIdentity account.Identity
-	clean        func()
-}
-
-func (test *repoBBTest) SetupTest() {
-	test.repo = space.NewRepository(test.DB)
-	test.clean = cleaner.DeleteCreatedEntities(test.DB)
-	testFxt := tf.NewTestFixture(test.T(), test.DB, tf.Identities(1))
-	test.testIdentity = *testFxt.Identities[0]
-}
-
-func (test *repoBBTest) TearDownTest() {
-	test.clean()
+	repo space.Repository
 }
 
-func (test *repoBBTest) TestCreate() {
-	res, _ := expectSpace(test.create(testSpace), test.requireOk)
-	require.Equal(test.T(), res.Name, testSpace)
-}
-
-func (test *repoBBTest) TestCreateFailSpaceNameChecks() {
-	expectSpace(test.create(""), test.assertBadParameter())
-}
-
-func (test *repoBBTest) TestCreateFailSameOwner() {
-	res, _ := expectSpace(test.create(testSpace), test.requireOk)
-
-	require.Equal(test.T(), res.Name, testSpace)
-
-	expectSpace(test.create(testSpace), test.assertDataConflict())
+func (s *repoBBTest) SetupSuite() {
+	s.DBTestSuite.SetupSuite()
+	s.repo = space.NewRepository(s.DB)
 }
 
-func (test *repoBBTest) TestLoad() {
-	expectSpace(test.load(uuid.NewV4()), test.assertNotFound())
-	res, _ := expectSpace(test.create(testSpace), test.requireOk)
-
-	res2, _ := expectSpace(test.load(res.ID), test.requireOk)
-	assert.True(test.T(), (*res).Equal(*res2))
-}
-
-func (test *repoBBTest) TestExistsSpace() {
-	t := test.T()
-	resource.Require(t, resource.Database)
-
-	t.Run("space exists", func(t *testing.T) {
-		// given
-		err := test.repo.CheckExists(context.Background(), space.SystemSpace.String())
+func (s *repoBBTest) TestCreate() {
+	s.T().Run("ok", func(t *testing.T) {
+		// given an identity
+		fxt := tf.NewTestFixture(t, s.DB, tf.Identities(1))
```

Comment:
Should we create that Identity in `SetupSuite`? Hence accessible via `s.Idn` in both tests below.

Reply:
I'd say no. I mean we could share state between tests but I also like the current level of test-isolation.